timesketch/api/v1/resources/contextlinks.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Context link API for version 1 of the Timesketch API."""
+import logging
+import re
+import copy
+
+from flask import jsonify
+from flask_restful import Resource
+from flask_login import login_required
+
+from timesketch.api.v1 import resources
+from timesketch.api.v1.utils import load_yaml_config
+
+# Set up logging
+logger = logging.getLogger("timesketch.contextlinks_api")
+
+
+class ContextLinkConfigResource(resources.ResourceMixin, Resource):
+    """Resource to get context link information."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+        Returns:
+            JSON object including version info
+        """
+        context_link_config = load_yaml_config("CONTEXT_LINKS_CONFIG_PATH")
+
+        response = {}
+
+        if not context_link_config:
+            return jsonify(response)
+
+        verification_results = []
+
+        # Check mandatory fields are correctly defined.
+        for item in context_link_config:
```

Comment:
Why we are validating the YAML file every time the API endpoint executes? This implies that we don't trust the supplied, server side, YAML file to follow the needed format.

Instead, we could add a validation function to tsctl that the server admin can run to validate the yaml file before it is used in the system. Then it is up to the admin to check that it has a valid file. We could then use something like jsonschema to validate the fields to simplify this logic.

WDYT? Let's discuss this in person.

Reply:
Thanks for the input. As discussed:

- Moved validation to tsctl for manual execution by an admin.
- Switched to jsonschema validation
- Removed validation from the API endpoint itself